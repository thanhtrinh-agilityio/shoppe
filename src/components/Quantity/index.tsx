import { AddIcon, MinusIcon } from '@chakra-ui/icons'
import { Box, IconButton, Text } from '@chakra-ui/react'
import { memo } from 'react'

interface QuantityProps {
  quantity: number
  onIncreaseCartQuantity: () => void
  onDecrementCartQuantity: () => void
}

const Quantity = ({
  quantity,
  onIncreaseCartQuantity,
  onDecrementCartQuantity,
}: QuantityProps) => {
  return (
    <Box
      background="gray"
      width="135px"
      display="flex"
      justifyContent="space-between"
      alignItems="center"
      borderRadius="4px"
    >
      <IconButton
        data-testid="increase-button"
        color="secondary"
        fontSize="x-small"
        onClick={onIncreaseCartQuantity}
        type="submit"
        aria-label="Search database"
        icon={<AddIcon />}
      />
      <Text color="secondary" fontSize="16px" data-testid="quantity">
        {quantity}
      </Text>
      <IconButton
        data-testid="decrement-button"
        color="secondary"
        fontSize="x-small"
        onClick={onDecrementCartQuantity}
        type="submit"
        aria-label="Search database"
        icon={<MinusIcon />}
      />
    </Box>
  )
}

export default memo(Quantity)
