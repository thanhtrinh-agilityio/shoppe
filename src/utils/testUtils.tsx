import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

// Themes
import { customTheme } from '@themes/index'

// Contexts
import { CartProvider } from '@contexts/CartProvider'

export const customRender = (component: ReactNode) => {
  return render(
    <ChakraProvider theme={customTheme}>
      <CartProvider>{component}</CartProvider>
    </ChakraProvider>,
  )
}

export * from '@testing-library/react'
export * from '@testing-library/jest-dom'
export { customRender as render }
