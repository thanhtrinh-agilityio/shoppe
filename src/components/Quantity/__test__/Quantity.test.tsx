// Component
import Quantity from '@components/Quantity'

// Utils
import { fireEvent, render, screen } from '@utils/testUtils'

describe('Quantity render', () => {
  const mockProps = {
    quantity: 1,
    onIncreaseCartQuantity: jest.fn(),
    onDecrementCartQuantity: jest.fn(),
  }
  it('Should show match Quantity DOM Snapshot', () => {
    const { container } = render(<Quantity {...mockProps} />)
    expect(container).toMatchSnapshot()
  })

  it('Should simulate onIncrease Cart Quantity event and expect mock function', () => {
    render(<Quantity {...mockProps} />)
    const increaseButton = screen.getByTestId('increase-button')
    expect(increaseButton).toBeTruthy()

    fireEvent.click(increaseButton)
    expect(mockProps.onIncreaseCartQuantity).toBeCalled()
  })

  it('Should simulate onDecrease Cart Quantity event and expect mock function', () => {
    render(<Quantity {...mockProps} />)
    const decreaseButton = screen.getByTestId('decrement-button')
    expect(decreaseButton).toBeTruthy()

    fireEvent.click(decreaseButton)
    expect(mockProps.onDecrementCartQuantity).toBeCalled()
  })
})
