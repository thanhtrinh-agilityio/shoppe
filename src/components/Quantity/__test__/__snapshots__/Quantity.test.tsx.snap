// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quantity render Should show match Quantity DOM Snapshot 1`] = `
<div>
  <div
    class="css-drsjqf"
  >
    <button
      aria-label="Search database"
      class="chakra-button css-rzypxw"
      data-testid="increase-button"
      type="submit"
    >
      <svg
        aria-hidden="true"
        class="chakra-icon css-onkibi"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
          fill="currentColor"
        />
      </svg>
    </button>
    <p
      class="chakra-text css-17i4smz"
      data-testid="quantity"
    >
      1
    </p>
    <button
      aria-label="Search database"
      class="chakra-button css-rzypxw"
      data-testid="decrement-button"
      type="submit"
    >
      <svg
        aria-hidden="true"
        class="chakra-icon css-onkibi"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g
          fill="currentColor"
        >
          <rect
            height="4"
            width="20"
            x="2"
            y="10"
          />
        </g>
      </svg>
    </button>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
